napari/utils/perf/_utils.py

```diff
@@ -81,29 +96,26 @@ def draw(self):
 
         def decorator(func):
             @functools.wraps(func)
-            def time_function(*args, **kwargs):
+            def wrapper(*args, **kwargs):
```

Comment:
I like the previous name better, as it's more descriptive... But maybe `timed_function`?

Reply:
Also, the multiple nesting can be avoided with `toolz.curry`, which we already depend on and use in the `thread_worker` decorator:

```python
@tz.curry
def perf_func_named(timer_name : str, func : Callable) -> Callable:
    @functools.wraps(func)
    def timed_func(*args, **kwargs):
        with perf_timer(timer_name, "decorator"):
            return func(*args, **kwargs)
    return timed_func
```

By the way, what is the second argument, 'decorator', to perf_timer?

Reply:
... And you could even combine things with the above non-named decorator:

```python
@tz.curry
def perf_func_named(func : Callable, timer_name : str = None) -> Callable:
    if timer_name is None:
        timer_name = f"{func.__qualname__}"

    @functools.wraps(func)
    def timed_func(*args, **kwargs):
        with perf_timer(timer_name, "decorator"):
            return func(*args, **kwargs)
    return timed_func
```

Then you would use

```python
func = perf_func(func, timer_name=my_name)
```

(using keyword arguments for `timer_name`) when wanting to overwrite the name, but just `func = perf_func(func)` when not.

I'm not sure what `full_name` is above but the same principle applies.

Reply:
The "decorator" string is the second argument of `perf_timer` which is "category", which is like a comma separated list of strings that supposed let you toggle on and off events by "category". Like "rendering" vs. "IO" maybe? I've never actually seen it in the UI. I should probably just have no category until/unless we determine it's useful for something.

I've never used `toolz.curry` so that would take some investigation, but sounds potentially useful. Same with collapsing it down to less decorators.

Reply:
I just deleted `perf_func` and `perf_func_named`. Those came long before we had the patcher. With the patcher there's really no reason to have these. The patcher patches in `perf_timers` it does not use these decorators. And patching using the patcher is really better in every way that using these decorators except you lose the "name" part. But we could actually add that to the patcher if we wanted.

That is the patcher config file could support something like:

    ("module1.Class2.method2", "redraw everything")

meaning patch `method2` to give the timer the name "redraw everything".

So if we ever need names like that I'd add them with the patcher.